api/osf_groups/views.py

```diff
@@ -0,0 +1,231 @@
+from django.apps import apps
+from django.db.models import Q
+
+from rest_framework import generics, permissions as drf_permissions
+from rest_framework.exceptions import NotFound, ValidationError
+
+from api.base import permissions as base_permissions
+from api.base.exceptions import InvalidFilterOperator, InvalidFilterValue
+from api.base.filters import ListFilterMixin
+from api.base.utils import get_object_or_error, get_user_auth, is_bulk_request
+from api.base.views import JSONAPIBaseView
+from api.base import generic_bulk_views as bulk_views
+from api.osf_groups.permissions import IsGroupManager, GroupMemberManagement
+from api.osf_groups.serializers import (
+    GroupSerializer,
+    GroupDetailSerializer,
+    GroupMemberSerializer,
+    GroupMemberDetailSerializer,
+    GroupMemberCreateSerializer,
+)
+from api.users.views import UserMixin
+from framework.auth.oauth_scopes import CoreScopes
+from osf.models import OSFGroup, OSFUser
+from osf.utils.permissions import MANAGER, GROUP_ROLES
+
+
+class OSFGroupMixin(object):
+    """
+    Mixin with convenience method for retrieving the current OSF Group
+    """
+    group_lookup_url_kwarg = 'group_id'
+
+    def get_osf_group(self, check_object_permissions=True):
+
+        group = get_object_or_error(
+            OSFGroup,
+            self.kwargs[self.group_lookup_url_kwarg],
+            self.request,
+            display_name='osf_group',
+        )
+
+        if check_object_permissions:
+            self.check_object_permissions(self.request, group)
+        return group
+
+
+class GroupList(JSONAPIBaseView, generics.ListCreateAPIView, ListFilterMixin, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    serializer_class = GroupSerializer
+    view_category = 'groups'
+    view_name = 'group-list'
+    ordering = ('-modified', )
+
+    def get_default_queryset(self):
+        user = self.request.user
+        if user.is_anonymous:
+            return OSFGroup.objects.none()
+        return user.osf_groups
+
+    # overrides ListCreateAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        """Create an OSFGroup.
+
+        :param serializer:
+        """
+        # On creation, logged in user is the creator
+        user = self.request.user
+        serializer.save(creator=user)
+
+
+class GroupDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, OSFGroupMixin):
```

Comment:
Again you might want to consolidate  these (`GroupDetail` and `GroupList`) to inherit from a base class, the static elements are very similar and future additions probably won't deviate from them much.

Reply:
ok, I try to weigh DRY versus readable, and if only a little code is consolidated, I don't find it's always worth it.  I can consolidate this, however.